Adds a new core integration imou_life

Imou-OpenPlatform opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adds a new core integration imou

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request:  https://github.com/home-assistant/home-assistant.io/pull/43104
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Imou-OpenPlatform committed:
Adds new integration imou_life.

Imou-OpenPlatform committed:
Merge branch 'home-assistant:dev' into imou_life

Imou-OpenPlatform committed:
Merge branch 'dev' into imou_life

jcisio commented on homeassistant/components/imou_life/README.md:
Who would read this file? I think it is ported from the custom integration. I'd merge this file into the doc PR since it now has its own page.

jcisio commented on homeassistant/components/imou_life/manifest.json:
```diff
@@ -0,0 +1,12 @@
+  "codeowners": ["@Imou-OpenPlatform"],
+  "config_flow": true,
+  "dependencies": [],
+  "documentation": "https://next.home-assistant.io/integrations/imou_life",
```
Not sure if you leave `next` here or the current url.

jcisio commented on homeassistant/components/imou_life/quality_scale.yaml:
```diff
@@ -0,0 +1,75 @@
+    status: exempt
+    comment: Integration does not register custom actions.
+  appropriate-polling: done
+  brands:
```
The brand is imou. Do you have separate brand for Imou life? If not I'd recommend to rename this integration to `imou` and put the brand assets under this name. This integration already supports multiple classes of device. If ever you create another imou integration, you can create an alias.
But in any case you still have to put assets in the brands repo so that the documentation could be rendered with the logo.

Imou-OpenPlatform committed:
Merge branch 'home-assistant:dev' into imou_life

Imou-OpenPlatform committed:
Rename imou_life integration to imou and update related files

- Rename component directory from imou_life to imou
- Update manifest.json, const.py, and other integration files
- Update test files and CODEOWNERS
- Remove README.md from old location
- Update requirements_all.txt

Imou-OpenPlatform committed:
Add missing data descriptions and simpleeval dependency to imou integration

- Add data_description fields for app_id, app_secret, and api_url in config flow
- Add data_description for rotation_duration in options flow
- Add simpleeval==1.0.3 to manifest requirements
- Format code and update generated files

This resolves translation errors in tests and adds required dependency for the integration.

Imou-OpenPlatform commented:
> Hello @Imou-OpenPlatform,
> 
> When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).
> 
> The commits that are missing a linked GitHub account are the following:
> 
> * [`3299ad57b6136434605b0659f65856e960268f64`](https://github.com/home-assistant/core/pull/159732/commits/3299ad57b6136434605b0659f65856e960268f64) - This commit has something that looks like an email address ([contributor@home-assistant.io](mailto:contributor@home-assistant.io)). Maybe try linking that to GitHub?.
> * [`26ae57291e8873dd35035f649dd94262c7ce37f2`](https://github.com/home-assistant/core/pull/159732/commits/26ae57291e8873dd35035f649dd94262c7ce37f2) - This commit has something that looks like an email address ([contributor@home-assistant.io](mailto:contributor@home-assistant.io)). Maybe try linking that to GitHub?.
> 
> Unfortunately, **we are unable to accept this pull request until this situation is corrected.**
> 
> Here are your options:
> 
> 1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).
> 2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
>    
>    * If you only made a single commit you should be able to run
>      ```
>      git commit --amend --author="Author Name <email@address.com>"
>      ```
>      
>      
>          
>            
>          
>      
>            
>          
>      
>          
>        
>      (substituting "Author Name" and "`email@address.com`" for your actual information) to set the authorship information.
>    * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
>      
>      1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
>      2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
>    * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.
> 
> We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.
> 
> Thanks, I look forward to checking this PR again soon! ❤️

Please check again.

Imou-OpenPlatform committed:
Merge branch 'home-assistant:dev' into imou_life
